exporter/views.py

```diff
@@ -120,34 +125,37 @@ def get(self, request, *args, **kwargs):
                 f.write(asset.latest_transcription or "")
 
         # Turn Structure into bagit format
-        bagit.make_bag(export_base_dir, {"Contact-Name": request.user.username})
+        bagit.make_bag(export_base_dir)
 
         # Build .zip file of bagit formatted Campaign Folder
         archive_name = export_base_dir
         shutil.make_archive(archive_name, "zip", export_base_dir)
 
-        # Download zip
-        with open("%s.zip" % export_base_dir, "rb") as zip_file:
-            response = HttpResponse(zip_file, content_type="application/zip")
-
-        export_filename = "%s-%s.zip" % (
-            campaign_slug,
-            datetime.today().isoformat(timespec="minutes"),
-        )
-
-        response["Content-Disposition"] = "attachment; filename=%s" % export_filename
-
-        # Upload zip to S3 bucket
-        s3_bucket = getattr(settings, "S3_BUCKET_NAME", None)
-
-        if s3_bucket:
-            s3 = boto3.resource("s3")
-            s3.Bucket(s3_bucket).upload_file(
-                "%s.zip" % export_base_dir, "exporter/%s" % export_filename
+        try:
+            # Upload zip to S3 bucket
+            s3_bucket = getattr(settings, "EXPORT_S3_BUCKET_NAME", None)
+            export_filename = "%s-%s.zip" % (
+                campaign_slug,
+                datetime.today().isoformat(timespec="minutes"),
             )
 
-        # Clean up temp folders & zipfile once exported
-        shutil.rmtree(export_base_dir)
-        os.remove("%s.zip" % export_base_dir)
-
-        return response
+            if s3_bucket:
+                s3 = boto3.resource("s3")
+                s3.Bucket(s3_bucket).upload_file(
+                    "%s.zip" % export_base_dir, "%s" % export_filename
+                )
+                # TODO: return a link to this file
+                # return redirect to the S3 object's
+                # public URL in the response returned by `upload_file`
+            else:
+                # Download zip
+                with open("%s.zip" % export_base_dir, "rb") as zip_file:
+                    response = HttpResponse(zip_file, content_type="application/zip")
+                response["Content-Disposition"] = (
+                    "attachment; filename=%s" % export_filename
+                )
+                return response
+        finally:
+            # Clean up temp folders & zipfile once exported
+            shutil.rmtree(export_base_dir)
```

Comment:
Since this is called within a `TemporaryDirectory` context it should be cleaned up automatically without us needing to do anything here.

Reply:
It looks like the last round of comments wasn't on the latest version of the code?

Reply:
They were current at the time I wrote them — not sure whether that overlapped with more changes

Reply:
Weird cause I could've sworn I took out the try / finally before I went to the doctor.